Migrated the existing Spring application from the previous module to …

Fat-Frumos opened the pull request:
…Spring Boot:

  - Updated dependencies in configuration files (pom.xml)
  - Removed AppConfig, RestConfig, WebAppInitializer
  - PersistenceConfig added
  - RestApiApplication starter was created.
  - java version updated to 1.8

Fat-Frumos committed:
Migrated the existing Spring application from the previous module to Spring Boot:
  - Updated dependencies in configuration files (pom.xml)
  - Removed AppConfig, RestConfig, WebAppInitializer
  - PersistenceConfig added
  - RestApiApplication starter was created.
  - java version updated to 1.8

Fat-Frumos committed:
Changed jdbcTemplate to entity Manager
Added Persistence annotation to entity
Created Model Assembler, Entity Generator

sslichnyi commented:
It will be good to create a new module 'starter' and put there main class with all properties (application.properties). And config package

sslichnyi commented:
please, move db sql files to model module

sslichnyi commented:
Please, remove spring-boot-starter-data-jpa dependency and use only hibernate

sslichnyi commented on Module#03/api/src/main/java/com/epam/esm/RestApiApplication.java:
```diff
@@ -0,0 +1,16 @@
+@SpringBootApplication
+@EntityScan(basePackages = "com.epam.esm.entity")
+@Import(PersistenceConfig.class)
+public class RestApiApplication {
```
RestApiApplication Should be moved

sslichnyi replied:
Should be moved to a new module

sslichnyi commented:
please, add IntelliJ Configuration files "Store as project file"

sslichnyi commented:
please, edit application files in discussed way.

application.properties:
spring.datasource.url=jdbc:h2:mem:db
spring.datasource.username=user
spring.datasource.password=pass
spring.datasource.driverClassName=org.h2.Driver
spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.H2Dialect
spring.jpa.hibernate.ddl-auto=create
spring.jpa.show-sql=false

spring.mvc.static-path-pattern=/resources/**
spring.mvc.throw-exception-if-no-handler-found=true

spring.main.allow-circular-references=true
spring.main.banner-mode=off

server.tomcat.accesslog.enabled=true

sslichnyi commented on Module#03/api/src/main/java/com/epam/esm/config/PersistenceConfig.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
```
public LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource)

Fat-Frumos replied:
Injected dependency dataSource:  
 - EntityManager Factory,  
 - DataSource Initializer;

sslichnyi commented on Module#03/api/src/main/java/com/epam/esm/config/PostgreSQLConfig.java:
```diff
@@ -26,7 +26,7 @@ public class PostgreSQLConfig {
     private String url;
 
-        private DataSource dataSource() {
+    private DataSource dataSource() {
```
This class is redundant. Please, delete this file.

sslichnyi commented on Module#03/api/src/main/java/com/epam/esm/config/H2Config.java:
```diff
@@ -1,34 +1,39 @@
+public class H2Config {
+    @Profile("test")
     public DataSource dataSource() {
         return new EmbeddedDatabaseBuilder()
```
Move this logic to PersistenceConfig class

sslichnyi commented on Module#03/api/src/main/java/com/epam/esm/controller/CertificateController.java:
```diff
@@ -1,78 +1,90 @@
 @RestController
 @RequiredArgsConstructor
-@RequestMapping(value = "/certificates")
+@RequestMapping(value = "/api/certificates")
```
'/api' - path can be applied the hole project with 'server.servlet.context-path=/api' configuration in application.properties file. Please take a look on it

Fat-Frumos replied:
config servlet.context path was added

Fat-Frumos committed:
Added server servlet context-path= /api configuration
Added folder .run  with contains 4 config files
Removed PostgreSQLConfig, H2JdbcConfig
Injected dependency dataSource:
 - EntityManager Factory,
 - DataSource Initializer;
LocalContainerEntityManagerFactoryBean in EntityManager
updated application.properties, data source driver H2 was changed
removed spring-boot-starter-data-jpa, added spring-jdbc, spring-tx, spring-orm

Fat-Frumos committed:
Added pagination
Activeted Entity Generator
Created tests for Criteria, FilterParams

sslichnyi commented:
Please, create the following structure:
modules:
- model (entities, dto)
- repository-api (custom repository interfaces)
- repository-impl (bean which implements repository-api's interfaces)
- service-api (custom server interfaces)
- service-impl (beans which implements service-api interfaces, exceptions)
- web (config, SpringBootApplication main class, properties, controllers, dtos, excpetion handler, exceptions)
